File: core/src/net/natruid/jungle/components/TileComponent.java
package net.natruid.jungle.components;

import com.artemis.Component;
import com.artemis.annotations.EntityId;
import net.natruid.jungle.utils.Point;
import net.natruid.jungle.utils.types.TerrainType;

public class TileComponent extends Component {
    public final Point coord = new Point();
    public TerrainType terrainType = TerrainType.NONE;
    public boolean hasRoad = false;
    @EntityId
    public int unit = -1;
    @EntityId
    public int obstacle = -1;

    protected void reset() {
        coord.set(0, 0);
        terrainType = TerrainType.NONE;
    }
}

File: core/src/net/natruid/jungle/components/TileComponent.kt
package net.natruid.jungle.components

import com.artemis.Component
import com.artemis.annotations.EntityId
import net.natruid.jungle.utils.Point
import net.natruid.jungle.utils.TerrainType

class TileComponent : Component() {
    val coord: Point = Point()
    var terrainType: TerrainType = TerrainType.NONE
    var hasRoad: Boolean = false
    @EntityId
    var unit: Int = -1
    @EntityId
    var obstacle: Int = -1

    override fun toString(): String {
        return coord.toString()
    }
}
